drizzle-orm/src/sql/expressions/conditions.ts

```diff
@@ -204,22 +393,116 @@ export function notInArray(
 	return sql`${column} not in ${bindIfParam(values, column)}`;
 }
 
+
+/**
+ * Test whether an expression is NULL. By the SQL standard,
+ * NULL is neither equal nor not equal to itself, so
+ * it's recommended to use `isNull` and `notIsNull` for
+ * comparisons to NULL.
+ *
+ * ## Examples
+ *
+ * ```ts
+ * // Select cars that have no discontinuedAt date.
+ * db.select(cars)
+ *   .where(
+ *     isNull(cars.discontinuedAt))
+ * ```
+ *
+ * @see isNotNull for the inverse of this test
+ */
 export function isNull(column: AnyColumn | Placeholder | SQLWrapper): SQL {
 	return sql`${column} is null`;
 }
 
+/**
+ * Test whether an expression is not NULL. By the SQL standard,
+ * NULL is neither equal nor not equal to itself, so
+ * it's recommended to use `isNull` and `notIsNull` for
+ * comparisons to NULL.
+ *
+ * ## Examples
+ *
+ * ```ts
+ * // Select cars that have been discontinued.
+ * db.select(cars)
+ *   .where(
+ *     isNotNull(cars.discontinuedAt))
+ * ```
+ *
+ * @see isNull for the inverse of this test
+ */
 export function isNotNull(column: AnyColumn | Placeholder | SQLWrapper): SQL {
 	return sql`${column} is not null`;
 }
 
+/**
+ * Test whether a subquery evaluates to have any rows.
+ *
+ * ## Examples
+ *
+ * ```ts
+ * // Users whose `homeCity` column has a match in a cities
+ * // table.
+ * db
+ *   .select()
+ *   .from(users)
+ *   .where(
+ *     exists(db.select()
+ *      .from(cities)
+ *      .where(eq(users.homeCity, cities.id))));
```

Comment:
should `exists` be here instead of `eq`?

Reply:
There's the `exists` on line 451 right above this - I think this query is right, it's adapted straight from https://github.com/tmcw/drizzle-orm/blob/58392e0cf087d0233b2c78d63a86a839a539d075/drizzle-orm/type-tests/sqlite/select.ts#L128-L131 with just whitespace changes.

Reply:
agree, just didn't see it